src/fontra/views/editor/editor.js

```diff
@@ -2845,6 +2883,57 @@ export class EditorController {
     }
   }
 
+  async glyphSearchDialog(
+    titleLabel = translate("dialog.glyphs.search"),
+    okLabel = translate("dialog.add")
+  ) {
+    const glyphsSearch = document.createElement("glyphs-search");
+    glyphsSearch.glyphMap = this.fontController.glyphMap;
+
+    glyphsSearch.addEventListener("selectedGlyphNameChanged", (event) => {
+      dialog.defaultButton.classList.toggle(
+        "disabled",
+        !glyphsSearch.getSelectedGlyphName()
+      );
+    });
+
+    glyphsSearch.addEventListener("selectedGlyphNameDoubleClicked", (event) => {
+      dialog.defaultButton.click();
+    });
+
+    const dialog = await dialogSetup(titleLabel, null, [
+      { title: translate("dialog.cancel"), isCancelButton: true },
+      { title: okLabel, isDefaultButton: true, resultValue: "ok", disabled: true },
+    ]);
+
+    dialog.setContent(glyphsSearch);
+
+    setTimeout(() => glyphsSearch.focusSearchField(), 0); // next event loop iteration
+
+    if (!(await dialog.run())) {
+      // User cancelled
+      return;
+    }
+
+    const glyphName = glyphsSearch.getSelectedGlyphName();
+    if (!glyphName) {
+      // Invalid selection
+      return;
+    }
+
+    return glyphName;
+  }
+
+  async doInsertGlyphToCanvas(titleLabel, okLabel, where) {
+    const glyphName = await this.glyphSearchDialog(titleLabel, okLabel);
+    if (!glyphName) {
+      return;
+    }
+    const glyphInfo = this.fontController.glyphInfoFromGlyphName(glyphName);
+    this.insertGlyphInfos([glyphInfo], where, true);
+    // TODO: How do we handle the undo/redo in that case?
```

Comment:
The browser's "go back" and "go forward" buttons/keys deal with this. We do not have undo here.

Reply:
ok, thanks. I removed the comment.